src/main/java/ch/njol/skript/registrations/Classes.java

```diff
@@ -68,6 +53,7 @@ private Classes() {}
 	private final static HashMap<Class<?>, ClassInfo<?>> exactClassInfos = new HashMap<>();
 	private final static HashMap<Class<?>, ClassInfo<?>> superClassInfos = new HashMap<>();
 	private final static HashMap<String, ClassInfo<?>> classInfosByCodeName = new HashMap<>();
+	private final static Map<String, List<ClassInfo<?>>> registeredPatterns = new HashMap<>();
```

Comment:
i have quite a few concerns with this approach
First, naming-wise it's very vague and doesn't specify that it's for literal patterns only.
Second, it's being handled on Classes instead of each classinfo storing their own patterns (or better yet, the parser itself storing the patterns it uses.)
Third, it doesn't handle conflicts between parsers that don't have defined patterns.

I think that a) this approach could be done better by doing it where the data is actually used and comes from (the parsers), and b) that the approach is quite limited and something better could be done, like a parse method that checks against all classinfos and automatically stores that info in UnparsedLiteral for better/faster reparsing and easy checks if it matches more than 1.